src/run.jl

```diff
@@ -16,14 +16,36 @@ function restarted_arnoldi(A::AbstractMatrix{T}, min = 5, max = 30, nev = min, 
     active = 1
     V_prealloc = Matrix{T}(undef, n, min)
     for restarts = 1 : max_restarts
+        n = max - active + 1
 
         iterate_arnoldi!(A, arnoldi, min′ + 1 : max, h)
         
         # Compute the eigenvalues of the active part
-        Q = Matrix{T}(I, max, max)
+        Q = Matrix{T}(I, n, n)
         H_copy = copy(view(arnoldi.H, active:max, active:max))
         local_schurfact!(H_copy, Q)
-        λs = sort!(eigvalues(H_copy), by = abs, rev = true)
+        # λs = sort!(eigvalues(H_copy), by = abs, rev = true)
+        λs = eigvalues(H_copy)
+
+        y = Vector{T}(undef,n)
+        res = Vector{Float64}(undef,n)
+        @inbounds for i = n : -1 : 1
+            R = copy(H_copy)
```

Comment:
No need to make a copy, just work with `H - lambda * I` implicitly; pass H and lambda to the backward substitution function.